### PR TITLE
Fix broken links in v2-vs-v3 doc

### PR DESCRIPTION
**Pull request by dbrakman**
# [DEVDOCS-](https://jira.bigcommerce.com/browse/DEVDOCS-)

Links like .../products-overview would redirect to .../catalog/catalog-overview but not preserve fragments.

## What changed?
* Changed mentions of `/products-overview` to `/catalog/catalog-overview` so that links like https://developer.bigcommerce.com/api-docs/store-management/catalog/catalog-overview#complex-rules take the reader to the appropriate section of the target doc.

### CODE REVIEW

**Comment by uwikaiddi** [comment]
@ektravel, could you review these changes since you're working with these docs?

File: docs/api-docs/catalog/v2-vs-v3.md [hunk 1]
```diff
@@ -124,11 +124,11 @@ In V3, any variants or modifier adjusters created with non-null core properties
 ## Related resources
 
 ### Articles
-* [Catalog Overview](https://developer.bigcommerce.com/api-docs/store-management/products-overview)
+* [Catalog Overview](https://developer.bigcommerce.com/api-docs/store-management/catalog/catalog-overview/)
 * [Deprecations and sunsets](https://developer.bigcommerce.com/api-reference#deprecations-and-sunsets)
```
Comment (ektravel) [inline docs/api-docs/catalog/v2-vs-v3.md:128]:
```suggestion
* [Deprecations and sunsets](https://developer.bigcommerce.com/api-docs/getting-started/deprecations-and-sunsets)
```

File: docs/api-docs/catalog/v2-vs-v3.md [hunk 2]
```diff
@@ -124,11 +124,11 @@ In V3, any variants or modifier adjusters created with non-null core properties
...
 
 ### Endpoints
 * [V3 Product Complex Rules](https://developer.bigcommerce.com/api-reference/store-management/catalog/product-complex-rules)
 * [V3 Product Options](https://developer.bigcommerce.com/api-reference/store-management/catalog/product-options)
 * [V3 Product Modifiers](https://developer.bigcommerce.com/api-reference/store-management/catalog/product-modifiers)
-* [V3 Product Variants](https://developer.bigcommerce.com/api-reference/catalog/catalog-api/product-variants)
\ No newline at end of file
+* [V3 Product Variants](https://developer.bigcommerce.com/api-reference/catalog/catalog-api/product-variants)
```
Comment (ektravel) [inline docs/api-docs/catalog/v2-vs-v3.md:134]:
```suggestion
* [V3 Product Variants](https://developer.bigcommerce.com/api-reference/store-management/catalog/product-variants)
```

**Review by ektravel — COMMENTED** [review]
Updated **V3 Product Variants** and **Deprecations and Sunsets** links.

**Review by uwikaiddi — APPROVED** [review]
Thank you, @dbrakman .

**Pull request merged**